fast64_internal/f3d/f3d_gbi.py

```diff
@@ -2809,13 +2833,152 @@ def save_binary(self, romfile, f3d, segments):
+    def inline_triGroup(self, f3d: F3D, triGroup: "FTriGroup", bleed: "BleedGfx"):
+        # add material
+        self.draw.commands.extend(bleed.bled_mats.commands)
+        # bleed tri group (for large textures)
+        self.processInlineTriGroup(f3d, triGroup, bleed)
+        # add textures
+        for tile, texGfx in enumerate(bleed.bled_tex):
+            self.draw.commands.extend(texGfx.commands)
+        # add in triangles
+        self.draw.commands.extend(triGroup.triList.commands)
+        
+    # process the triGroup cmds
+    def processInlineTriGroup(self, f3d: F3D, triGroup: "FTriGroup", bleed: "BleedGfx"):
+        if triGroup.fMaterial.useLargeTextures:
+            usage_dict = dict()
+            cnt = 0
+            tri_commands = copy.copy(triGroup.triList.commands)  # copy the commands
+            for j, cmd in enumerate(tri_commands):
+                if not cmd.bleed([], [], 0):
+                    continue
+                last_use = usage_dict.get((type(cmd), getattr(cmd, "tile", None)), None)
+                usage_dict[(type(cmd), getattr(cmd, "tile", None))] = cmd
+                if last_use == cmd:
+                    triGroup.triList.commands.pop(j - cnt)  # list gets smaller as I pop, so modify index by num popped
+                    cnt += 1
+        return
+
+    def onBleedStart(self, f3d: F3D, lastMat: "FMaterial" = None):
+        # remove SPDisplayList from FMesh.draw
+        iter_cmds = copy.copy(self.draw.commands)
+        spDLCmds = (c for c in iter_cmds if type(c) == SPDisplayList)
+        for spDL in spDLCmds:
+            self.draw.commands.remove(spDL)
+        return
+
+    def onTriGroupBleedEnd(self, f3d: F3D, triGroup: "FTriGroup", lastMat: "FMaterial", bleed: "BleedGfx"):
+        return
+
+    def onBleedEnd(self, f3d: F3D, lastMat: "FMaterial"):
+        # revert certain cmds for extra safety
+        self.draw.commands.extend(
+            [
+                DPPipeSync(),
+                SPGeometryMode(["G_TEXTURE_GEN"], ["G_LIGHTING"]),
+                DPSetCycleType("G_CYC_1CYCLE"),
+                DPSetTextureLUT("G_TT_NONE"),
+            ]
+        )
+        return
+
     def to_c(self, f3d, gfxFormatter):
         staticData = CData()
         if self.cullVertexList is not None:
             staticData.append(self.cullVertexList.to_c())
-        for triGroup in self.triangleGroups:
-            staticData.append(triGroup.to_c(f3d, gfxFormatter))
-        dynamicData = gfxFormatter.drawToC(f3d, self.draw)
+        if self.inline:
+            for triGroup in self.triangleGroups:
+                # add in vertex data
+                staticData.append(triGroup.vertexList.to_c())
+            staticData.append(gfxFormatter.drawToC(f3d, self.draw))
+            dynamicData = CData()
+        else:
+            for triGroup in self.triangleGroups:
+                staticData.append(triGroup.to_c(f3d, gfxFormatter))
+            dynamicData = gfxFormatter.drawToC(f3d, self.draw)
```

Comment:
I don't think editing thd tri list itself is a good idea because the tri list is used multiple times when things are referenced several times (like in armatures).
I could add a BleedTag to the triList gfxList though.